### PR TITLE
Parametrize ports and host with environment variables

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: src/test/scala-2.12/org/apache/james/gatling/simulation/Configuration.scala [hunk 1]
```diff
@@ -3,12 +3,13 @@ package org.apache.james.gatling.simulation
...
 import scala.concurrent.duration._
+import scala.util.Properties
 
 object Configuration {
 
-  val ServerHostName = "127.0.0.1"
-  val BaseJmapUrl = s"http://$ServerHostName:1080"
-  val BaseJamesWebAdministrationUrl = new URL(s"http://$ServerHostName:8000")
+  val ServerHostName = Properties.envOrElse("TARGET_HOSTNAME", "localhost")
+  val BaseJmapUrl = s"http://$ServerHostName:${Properties.envOrElse("JMAP_PORT", "1080")}"
```
Comment (chibenwa) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/Configuration.scala]: Why don't we extract the port also in a separated variable?

File: src/test/scala-2.12/org/apache/james/gatling/simulation/Configuration.scala [hunk 2]
```diff
@@ -3,12 +3,13 @@ package org.apache.james.gatling.simulation
...
+  val BaseJamesWebAdministrationUrl = new URL(s"http://$ServerHostName:${Properties.envOrElse("WEBADMIN_PORT", "8000")}")
```
Comment (chibenwa) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/Configuration.scala]: Idem for webadminPort

**Review by chibenwa — APPROVED** [review]

**Review by mbaechler — APPROVED** [review]

File: src/main/scala-2.12/org/apache/james/gatling/smtp/SmtpProtocol.scala [hunk 3]
```diff
@@ -24,12 +26,13 @@ object SmtpProtocol {
     override def onExit: Session => Unit = s => ()
   }
 
-  val DEFAULT_PORT = 25
-  val DEFAULT_PORT_SSL = 465
+  val PORT = Properties.envOrElse("SMTP_PORT", "25").toInt
```
Comment (remk) [inline src/main/scala-2.12/org/apache/james/gatling/smtp/SmtpProtocol.scala:29]: 587 to be able to test using authentication ?

Comment (blackheaven) [reply src/main/scala-2.12/org/apache/james/gatling/smtp/SmtpProtocol.scala:29]: I'm only focusing on parametrizing it, not changing underlying behavior

**Review by remk — APPROVED** [review]

**Review by rouazana — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**